Kitodo/src/main/java/de/sub/goobi/forms/RulesetForm.java

```diff
@@ -156,4 +152,47 @@ public void setRulesetId(int id) {
     public int getRulesetId() {
         return this.rulesetId;
     }
+
+    // TODO:
+    // replace calls to this function with "/pages/rulesetEdit" once we have
+    // completely switched to the new frontend pages
+    private String redirectToEdit(String urlSuffix) {
+        try {
+            String referrer = FacesContext.getCurrentInstance().getExternalContext().getRequestHeaderMap()
+                    .get("referer");
+            String callerViewId = referrer.substring(referrer.lastIndexOf("/") + 1);
+            if (!callerViewId.isEmpty() && callerViewId.contains("projects.jsf")) {
+                return "/pages/rulesetEdit" + urlSuffix;
+            } else {
+                return "/pages/RegelsaetzeBearbeiten" + urlSuffix;
+            }
+        } catch (NullPointerException e) {
+            // This NPE gets thrown - and therefore must be caught - when "RulesetForm" is
+            // used from it's integration test
+            // class "RulesetFormIT", where no "FacesContext" is available!
+            return "/pages/ProjekteBearbeiten" + urlSuffix;
```

Comment:
This looks wrong. Why should edit page for projects be returned if we are in ruleset editing?

Reply:
See answer in https://github.com/kitodo/kitodo-production/pull/1194: in the new layout, the projects page has several tabs, including rulesets and dockets. So several "edit" pages will have a link to the "projects" page in their breadcrumbs, even if they edit something else instead of projects.

Reply:
See the travis failing report:
```
Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.018 sec <<< FAILURE! - in de.sub.goobi.forms.RulesetFormIT
shouldRedirectAtRulesetCreation(de.sub.goobi.forms.RulesetFormIT)  Time elapsed: 0.009 sec  <<< FAILURE!
org.junit.ComparisonFailure: Should redirect to RegelsaetzeBearbeiten expected:</pages/[Regelsaetz]eBearbeiten?faces-re...> but was:</pages/[Projekt]eBearbeiten?faces-re...>
	at de.sub.goobi.forms.RulesetFormIT.shouldRedirectAtRulesetCreation(RulesetFormIT.java:38)
```
I did not write this without reason.

Reply:
This is actually a copy/paste error. `/pages/RegelsaetzeBearbeiten`should be returned.

Reply:
Sorry, your conversation is to fast, I accidentially deleted my comment.

Reply:
Please, do not delete your comments next time as then other comments must be deleted too. Thank you.